lib/os.js

```diff
@@ -82,31 +82,29 @@ const getNetworkInterfacesDepMsg =
   'os.getNetworkInterfaces is deprecated. Use os.networkInterfaces instead.';
 
 const avgValues = new Float64Array(3);
-const cpuValues = new Float64Array(6 * pushValToArrayMax);
 
 function loadavg() {
   getLoadAvg(avgValues);
   return [avgValues[0], avgValues[1], avgValues[2]];
 }
 
-function addCPUInfo() {
-  for (var i = 0, c = 0; i < arguments.length; ++i, c += 6) {
-    this[this.length] = {
-      model: arguments[i],
-      speed: cpuValues[c],
+function cpus() {
+  const data = getCPUs();
+  const result = [];
+  for (var i = 0; i < data.length; i += 7) {
```

Comment:
But could you write a few words about the data structure contract, and why 7?

Reply:
Shouldn't the source of truth be in the C++'s side? And the `i * 7` equations there explains things pretty well IMO.

Reply:
Ack.